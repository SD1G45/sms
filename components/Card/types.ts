export interface CardProps extends React.HTMLAttributes<HTMLDivElement> {
  className?: string;
}
