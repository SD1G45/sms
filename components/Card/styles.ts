import React from "react";
import styled from "styled-components";

export const StyledCard = styled.div`
  padding: 20px;
  box-shadow: 0px 3px 18px 5px rgba(0, 0, 0, 0.14);
  background-color: white;
`;
