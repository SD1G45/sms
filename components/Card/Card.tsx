import React from "react";
import { StyledCard } from "./styles";
import { CardProps } from "./types";

const Card = React.forwardRef<HTMLDivElement, CardProps>(
  ({ className, children }, ref) => {
    return (
      <StyledCard className={className} ref={ref}>
        {children}
      </StyledCard>
    );
  }
);

Card.displayName = "Card";
export default Card;
