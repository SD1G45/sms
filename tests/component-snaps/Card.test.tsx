import React from "react";
import renderer from "react-test-renderer";
import "jest-styled-components";
import Card from "../../components/Card";

describe("<Card/>", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Card/>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});