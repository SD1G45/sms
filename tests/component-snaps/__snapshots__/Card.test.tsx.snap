// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card/> renders correctly 1`] = `
.c0 {
  padding: 20px;
  box-shadow: 0px 3px 18px 5px rgba(0,0,0,0.14);
  background-color: white;
}

<div
  className="c0"
/>
`;
